web/client-api/src/main/java/io/deephaven/web/client/api/tree/JsTreeTable.java

```diff
@@ -209,455 +235,459 @@ public TreeRow(int offsetInSnapshot, Object[] dataColumns, Object rowStyleColumn
+                    viewRequest.setExpansions(expansions);
+                }
+                connection.hierarchicalTableServiceClient().view(viewRequest, connection.metadata(), c::apply);
+                return null;
+            }, error -> {
+                c.apply(error, null);
+                return null;
+            });
+        }));
+        return viewTicket;
     }
 
-    private void handleUpdate(Column[] columns, List<Sort> nextSort, List<FilterCondition> nextFilters,
-            TreeTableResult result, boolean alwaysFireEvent) {
-        JsLog.debug("tree table response arrived", result);
-        lastResult = result;
-        if (closed) {
-            if (viewportUpdateTimeoutId != null) {
-                DomGlobal.clearTimeout(viewportUpdateTimeoutId);
+    private void replaceSubscription() {
+        Promise<BiDiStream<?, ?>> stream = Promise.resolve(defer())
+                .then(ignore -> {
+                    makeKeyTable();
+                    TicketAndPromise filter = prepareFilter();
+                    TicketAndPromise sort = prepareSort(filter);
+                    TicketAndPromise view = makeView(sort);
+                    return Promise.all(
+                            keyTable,
+                            filter.promise,
+                            sort.promise,
+                            view.promise);
+                })
+                .then(results -> {
+                    BitSet columnsBitset = makeColumnSubscriptionBitset();
+                    RangeSet range = RangeSet.ofRange((long) (double) firstRow, (long) (double) lastRow);
+
+                    Column[] queryColumns = this.columns;
+
+                    boolean alwaysFireEvent = this.alwaysFireNextEvent;
+                    this.alwaysFireNextEvent = false;
+
+                    JsLog.debug("Sending tree table request", this,
+                            LazyString.of(() -> widget.getTicket().getTicket_asB64()),
+                            columnsBitset,
+                            range,
+                            alwaysFireEvent);
+                    BiDiStream<FlightData, FlightData> doExchange =
+                            connection.<FlightData, FlightData>streamFactory().create(
+                                    headers -> connection.flightServiceClient().doExchange(headers),
+                                    (first, headers) -> connection.browserFlightServiceClient().openDoExchange(first,
+                                            headers),
+                                    (next, headers, c) -> connection.browserFlightServiceClient().nextDoExchange(next,
+                                            headers,
+                                            c::apply),
+                                    new FlightData());
+
+                    FlightData subscriptionRequestWrapper = new FlightData();
+                    Builder doGetRequest = new Builder(1024);
```

Comment:
Why 1024? Maybe named const to make it clear why that number picked

Reply:
Ah, all the flatbuffer builders across the codebase are 1024, because that's what the docs show. Apparently this is explicit in other languages too, the api is pretty much the same in all languages.

https://google.github.io/flatbuffers/flatbuffers_guide_tutorial.html